src/components/button/Button.d.ts

```diff
@@ -1,11 +1,12 @@
 import React = require("react");
+import {Omit} from "../util";
 
-interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
+interface ButtonProps extends Omit<React.HTMLProps<HTMLButtonElement>, 'ref'> {
```

Comment:
This one is tricky. `React.HTMLProps<HTMLButtonElement>` already has a `ref`-prop which exposes the HTMLButtonElement. However `Button` (as every component) has its own `ref`. Those definitions now clash, so I removed it from the inherited props type. Also, the ref is now for the Button instance, not the DOM element. So if you need to expose the native element, there are two solutions: either have an extra prop that is passed to the DOM element (e.g. nativeRef) or consider using forward refs (https://reactjs.org/docs/forwarding-refs.html#forwarding-refs-to-dom-components).

This change was necessary for the DataScroller demo.